packages/flutter/lib/src/services/spell_check.dart

```diff
@@ -0,0 +1,553 @@
+            delegate: delegate,
+            endpoints: endpoints,
+            globalEditableRegion: globalEditableRegion,
+            selectionMidpoint: selectionMidpoint,
+            textLineHeight: textLineHeight,
+            spellCheckerSuggestionSpans: spellCheckResults,
+            );
+        }
+
+   // Provides a generous guesss of the spell check results for the current text if the spell check results for this text has not been received by the framework yet.
+   // Assumes: [1] order of results matches that of the text, [2] only a shift/deletion occurs at a time (this is verifiable)
+   List<SpellCheckerSuggestionSpan> correctSpellCheckResults(String newText, String resultsText, List<SpellCheckerSuggestionSpan> results) {
+       List<SpellCheckerSuggestionSpan> correctedSpellCheckResults = <SpellCheckerSuggestionSpan>[];
+
+       int span_pointer = 0;
+       bool foundBadSpan = false;
+
+       SpellCheckerSuggestionSpan currentSpan;
+       String oldSpanText;
+       String newSpanText;
+       int spanLength = 0;
+       int newStart = 0;
+       int searchStart = 0;
+
+       int? start_index;
+       int? end_index;
+       bool spanWithinTextRange = true;
+
+       int currentSpanStart = 0;
+       int currentSpanEnd = 0;
+
+       while(span_pointer < results.length) {
+           currentSpan = results[span_pointer];
+           currentSpanStart = currentSpan.start;
+           currentSpanEnd = currentSpan.end;
+
+           start_index = currentSpanStart < newText.length ? currentSpanStart : null;
+           end_index = currentSpanEnd < newText.length ? currentSpanEnd : null;
+
+           spanWithinTextRange = start_index != null && end_index != null;
+
+           if (!spanWithinTextRange) {
+               // No more of the spell check results will be within the range of the text
+               break;
+           } 
+           else {
+              oldSpanText = resultsText.substring(currentSpanStart, currentSpanEnd + 1);
+              newSpanText = newText.substring(currentSpanStart, currentSpanEnd + 1);
+
+                if (oldSpanText == newSpanText) {
+                    searchStart = currentSpanEnd + 1;
+                    correctedSpellCheckResults.add(currentSpan);
+                }
+                else {
+                    spanLength = currentSpanEnd - currentSpanStart; 
+                    RegExp regex = RegExp('\\b$oldSpanText\\b');
+                    int substring = newText.substring(searchStart).indexOf(regex);
+                    newStart = substring + searchStart;
+
+                    if (substring >= 0) {
+                        correctedSpellCheckResults.add(SpellCheckerSuggestionSpan(newStart, newStart + spanLength, currentSpan.replacementSuggestions));
+                        searchStart = newStart + spanLength;
+                    }
+                }
+           }
+
+           span_pointer += 1;
+       }
+
+       return correctedSpellCheckResults;
+   }
```

Comment:
Is there ever a case where it would be better to try to apply the old results to newText instead of the new ones?  I guess that would have to be like if you paste a totally new string and then immediately ctrl-z.  You wouldn't want the new results for the string that got pasted in, you would want the results you already had.

Maybe that's not very practical, but something to think about.

Reply:
Given that the Android engine doesn't try to give spell check results for all three (technically) strings but just for the old text and the re-pasted version of this old text, this case would be handled since the text that the old results correspond to would be equivalent to the new text.  

The case that wouldn't be handled is if the Android engine did try to give spell check results for all three, in which case, it would be hard to detect this scenario unless we kept a history of text.